### DISCUSSION TITLE
Why Timeout Exception has "Operation cancelled by user." even though user didn't do anything ?

### DISCUSSION THREAD

**Discussion by ActyVijay** [dotnet/SqlClient#2491] [Q&A]
I am getting the following SQL Exception in an ASP.NET app in production environment :- 
"Execution Timeout Expired.  The timeout period elapsed prior to completion of the operation or the server is not responding. Operation cancelled by user.".
Why am I getting "Operation cancelled by user" as I don't get this in my debug environment neither there is a way to stop query execution manually?
My query does have a timeout time which I just kept 5 sec to throw the timeout error in my debug environment? Please help why "Operation cancelled by user." is shown ?

**Comment by paulmedynski** [comment]
The string "Operation cancelled by user" is appended to certain exceptions when an operation the user (i.e. the app) has asked SqlClient to perform has timed out.  So here, the term "user" means you - the application using SqlClient.  Applications may configure timeouts for each operation, so when SqlClient is waiting for an operation to complete, and the user-supplied timeout has elapsed, SqlClient considers the operation to have been cancelled by the user, since the user asked SqlClient to cancel it after a period of time.
